core/arch/arm/mm/core_mmu.c

```diff
@@ -417,34 +408,6 @@ struct mobj **core_sdp_mem_create_mobjs(void)
 	}
 	return mobj_base;
 }
-
-static void check_sdp_intersection_with_nsec_ddr(void)
-{
-	const struct core_mmu_phys_mem *sdp_start =
-		&__start_phys_sdp_mem_section;
-	const struct core_mmu_phys_mem *sdp_end = &__end_phys_sdp_mem_section;
-	const struct core_mmu_phys_mem *ddr_start =
-		&__start_phys_nsec_ddr_section;
-	const struct core_mmu_phys_mem *ddr_end = &__end_phys_nsec_ddr_section;
-	const struct core_mmu_phys_mem *sdp;
-	const struct core_mmu_phys_mem *nsec_ddr;
-
-	if (sdp_start == sdp_end || ddr_start == ddr_end)
-		return;
-
-	for (sdp = sdp_start; sdp < sdp_end; sdp++) {
-		for (nsec_ddr = ddr_start; nsec_ddr < ddr_end; nsec_ddr++) {
-			if (core_is_buffer_intersect(sdp->addr, sdp->size,
-					     nsec_ddr->addr, nsec_ddr->size)) {
-				MSG_MEM_INSTERSECT(sdp->addr, sdp->size,
-						   nsec_ddr->addr,
-						   nsec_ddr->size);
-				panic("SDP <-> NSEC DDR memory intersection");
-			}
-		}
-	}
-}
```

Comment:
One than defines its dyn shm areas using `register_nsec_ddr()` (not DT or `register_ddr()`) will still need this to check they don't intersect the other memories (including SDP ones).

Reply:
Yes, it should not be deleted.